src/WebJobs.Script/Rpc/MessageExtensions/RpcMessageConversionExtensions.cs

```diff
@@ -102,40 +102,39 @@ public static TypedData ToRpc(this object value)
                     string rawBody = null;
 
                     MediaTypeHeaderValue mediaType = null;
-                    if (MediaTypeHeaderValue.TryParse(request.ContentType, out mediaType))
+                    MediaTypeHeaderValue.TryParse(request.ContentType, out mediaType);
+
+                    if (string.Equals(mediaType.MediaType, "application/json", StringComparison.OrdinalIgnoreCase))
```

Comment:
You can now get null refs here if the content type didn't parse. That's why I had this code in the if block previously.

I think you just need to add the missing else to the if/elseif I added, one that does the default parse/assignment that we're missing in the case when the content type DOES parse, but doesn't match one of our explicitly handled types.

Reply:
Also, please add a test for this - one that fails before the fix. I realize we have test debt here and the originator of the code didn't add any unit tests. But we need them so we can start stabilizing this and we got to pay the tax sometime :)

Should be easy to add some unit tests exercising these various conversion codepaths

Reply:
@mathewc and @fabiocav sounds good, thanks for the feedback :)

I'm having trouble with adding tests because I'm not sure how to mock an HttpRequest object? This might be really easy, but if either of you could help me out with that, that would be great!

Reply:
We should have examples of that and helpers in the `HttpTestHelpers`

Reply:
awesome, thank you!